roles/blackbox_exporter/tasks/configure.yml

```diff
@@ -48,3 +48,17 @@
     - blackbox_exporter
     - configure
     - blackbox_exporter_configure
+
+- name: Configure blackbox_exporter web
+  ansible.builtin.copy:
+    content: "{{ blackbox_exporter_web_config | to_nice_yaml(indent=2, sort_keys=False) }}"
+    dest: "{{ blackbox_exporter_config_dir }}/web.yml"
+    owner: "{{ blackbox_exporter_system_user }}"
+    group: "{{ blackbox_exporter_system_group }}"
+    mode: '0644'
+  notify:
+    - reload blackbox_exporter
+  tags:
+    - blackbox_exporter
+    - configure
+    - blackbox_exporter_configure
```

Comment:
```suggestion
```

No need to do this here, the `_common` role can take care of it, only thing you need to do is to pass the following vars to the `Common configure` task 

```yaml
_common_tls_server_config: "{{ blackbox_exporter_web_config.tls_server_config }}"
_common_http_server_config: "{{ blackbox_exporter_web_config.http_server_config }}"
_common_basic_auth_users: "{{ blackbox_exporter_web_config.basic_auth_users }}"
```